dev/sg/sg_insights.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"encoding/base64"
+	"strings"
+
+	"github.com/keegancsmith/sqlf"
+	"github.com/urfave/cli/v2"
+
+	"github.com/sourcegraph/log"
+
+	"github.com/sourcegraph/sourcegraph/dev/sg/internal/sgconf"
+	"github.com/sourcegraph/sourcegraph/dev/sg/internal/std"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	connections "github.com/sourcegraph/sourcegraph/internal/database/connections/live"
+	"github.com/sourcegraph/sourcegraph/internal/database/postgresdsn"
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+var (
+	insightsCommand = &cli.Command{
+		Name:     "insights",
+		Usage:    "Tools to interact with Code Insights data",
+		Category: CategoryUtil,
+		Subcommands: []*cli.Command{
+			{
+				Name:        "decode-id",
+				Usage:       "Decodes an encoded insight ID found on the frontend into an insight_view_id",
+				Description: `Run 'sg insights decode-id' to decode 1+ frontend IDs which can then be used for SQL queries`,
+				Action:      decodeInsightIDAction,
+			},
+			{
+				Name:        "series-id",
+				Usage:       "Gets all insight series ID from the base64 encoded frontend ID",
+				Description: `Run 'sg insights series-id' to decode a frontend ID and find all related series IDs`,
+				Action:      getInsightSeriesIDAction,
+			},
+		},
+	}
+)
+
+func decodeInsightIDAction(cmd *cli.Context) error {
+	ids := cmd.Args().Slice()
+	if len(ids) == 0 {
+		return errors.New("Expected at least 1 id to decode")
+	}
+	std.Out.WriteNoticef("Decoding %d IDs into insight_view_ids", len(ids))
+	for _, id := range ids {
+		cleanDecoded, err := decodeIDIntoSeriesID(id)
+		if err != nil {
+			return err
+		}
+		std.Out.Writef("\t%s -> %s", id, cleanDecoded)
+	}
+	return nil
+}
+
+func getInsightSeriesIDAction(cmd *cli.Context) error {
+	ids := cmd.Args().Slice()
+	if len(ids) != 1 {
+		return errors.New("Expected 1 id to decode")
+	}
+	std.Out.WriteNoticef("Finding all insight_series_ids for %s", ids[0])
```

Comment:
```suggestion
	std.Out.WriteNoticef("Finding all series IDs for %s", ids[0])
```
Or is it important to know `insight_series_ids` ?